vespajlib/src/main/java/com/yahoo/geo/OneDegreeParser.java

```diff
@@ -0,0 +1,282 @@
+     *
+     **/
+    public OneDegreeParser(boolean assume_n_s, String lat_or_lon) throws IllegalArgumentException {
+        this.parseString = lat_or_lon;
+        this.len = parseString.length();
+        consumeString(assume_n_s);
+    }
+
+    private void consumeString(boolean assume_n_s)  throws IllegalArgumentException {
+        char ch = getNextChar();
+
+        double degrees = 0.0;
+        double minutes = 0.0;
+        double seconds = 0.0;
+        boolean degSet = false;
+        boolean minSet = false;
+        boolean secSet = false;
+        boolean dirSet = false;
+        boolean foundDot = false;
+        boolean foundDigits = false;
+
+        boolean findingLatitude = false;
+        boolean findingLongitude = false;
+
+        double sign = +1.0;
+
+        int lastpos = -1;
+
+        // sign must be first character in string if present:
+        if (ch == '+') {
+            // unary plus is a nop
+            ch = getNextChar();
+        } else if (ch == '-') {
+            sign = -1.0;
+            ch = getNextChar();
+        }
+        do {
+            // did we find a valid char?
+            boolean valid = false;
+            if (pos == lastpos) {
+                throw new RuntimeException("internal logic error at '"+parseString+"' pos:"+pos);
+            } else {
+                lastpos = pos;
+            }
+
+            // first, see if we can find some number
+            double accum = 0.0;
+
+            if (isDigit(ch) || ch == '.') {
+                valid = true;
+                if (foundDigits) {
+                    throw new IllegalArgumentException("found digits after not consuming previous digits");
+                }
+                double divider = 1.0;
+                foundDot = false;
+                while (isDigit(ch)) {
+                    foundDigits = true;
+                    accum *= 10;
+                    accum += (ch - '0');
+                    ch = getNextChar();
+                }
+                if (ch == '.') {
+                    foundDot = true;
+                    ch = getNextChar();
+                    while (isDigit(ch)) {
+                        foundDigits = true;
+                        accum *= 10;
+                        accum += (ch - '0');
+                        divider *= 10;
+                        ch = getNextChar();
+                    }
+                }
+                if (!foundDigits) {
+                    throw new IllegalArgumentException("just a . is not a valid number");
+                }
+                accum /= divider;
+            }
+
+            // next, did we find a separator after the number?
+            // degree sign is a separator after degrees, before minutes
+            if (ch == '\u00B0' || ch == 'o') {
+                valid = true;
+                if (degSet) {
+                    throw new IllegalArgumentException("degrees sign only valid just after degrees");
```

Comment:
Consider adding the input string to all exception messages.

Reply:
done.